src/modules/schema.js

```diff
@@ -0,0 +1,189 @@
+   * signing key/capability is not present and the signature is received from outside.
+   * Repeatedly calling it will keep resetting the `signature` key.
+   * The signature must be one of the supported objects
+   * @param {object} signature - The schema's signature
+   */
+  setSignature(signature) {
+    if (signature instanceof Signature) {
+      this.signature = signature;
+    } else {
+      throw new Error('Provided signature object is not of instance Signature');
+    }
+
+    return this;
+  }
+
+  /**
+   * Serializes the object using `getSerializedBlob` and then signs it using the given
+   * polkadot-js pair. The object will be updated with key `signature`. Repeatedly calling it will
+   * keep resetting the `signature` key
+   * @param {any} msg - The message to sign
+   * @param {object} pair - Key pair to sign with
+   */
+  sign(msg, pair) {
+    this.signature = getSignatureFromKeyringPair(pair, msg);
+    return this;
+  }
+
+  /**
+   * Serializes to JSON for sending to the node, as `Blob`. A full DID is converted to the
+   * hex identifier and signature object is converted to the enum
+   * @returns {object}
+   */
+  toJSON() {
+    const {
+      signature,
+      id,
+      ...rest
+    } = this;
+
+    return {
+      ...rest,
+      id: getHexIdentifierFromBlobID(this.id),
+      signature: signature.toJSON(),
+    };
+  }
+
+  /**
+   * Serializes the schema to a blob object to send to the node
+   * @returns {object}
+   */
+  toBlob(author) {
+    return {
+      id: this.id,
+      blob: stringToHex(JSON.stringify(this.toJSON())),
+      author: getHexIdentifierFromDID(author),
+    };
+  }
+
+  /**
+   * Check that the given JSON schema is compliant with JSON schema spec mentioned in RFC
+   * @param {object} json - The JSON schema to validate
+   * @returns {Promise<object>} - Returns promise to an object or throws error
+   */
+  static async validateSchema(json) {
+    // Get the JSON schema spec to check against.
+    const jsonSchemaSpec = await this.getJSONSchemaSpec(json);
+    return validate(json, jsonSchemaSpec, {
+      throwError: true,
+    });
+  }
+
+  /**
+   * Get schema from from the chain using the given id, by querying the blob storage.
+   * Accepts a full blob id like blob:dock:0x... or just the hex identifier and the `DockAPI` object.
+   * The returned schema would be formatted as specified in the RFC (including author DID, schema id) or an error is
+   * returned if schema is not found on the chain or in JSON format.
+   * @param {string} id - The Schema ID
+   * @param {object} dockApi - The Dock API
+   * @returns {Promise<object>}
+   */
+  static async getSchema(id, dockApi) {
+    const hexId = getHexIdentifierFromBlobID(id);
+    const chainBlob = await dockApi.blob.getBlob(hexId);
+    const blobStr = u8aToString(chainBlob[1]);
+    try {
+      const schema = JSON.parse(blobStr);
+      schema.id = id;
+      schema.author = blobHexIdToQualified(u8aToHex(chainBlob[0]));
```

Comment:
author is not blob.

Reply:
you told me to change this in your last bit of feedback, look at what this method does

Reply:
My bad. I wrote blob:dock but should have written did:dock. The author is a DID.

Reply:
thought was a bit odd, but in this case doesnt that limit us to supporting only dock dids in this class? is that what we want?

Reply:
just thinking in terms of interop with other chains/resolvers

Reply:
Its getting schema from the dock chain so the DID will be of Dock

Reply:
makes sense, i wasnt sure if our chain-side supported other did methods aswell by way of just input/output